### PR TITLE
Use image/png on Wayland to improve copy support

### PR DESCRIPTION
**Pull request by nullobsi**
uses image/png only when wayland is detected on linux. works around a Qt bug where all images formats declared by Qt return an image in BMP format

### CODE REVIEW

**Comment by nullobsi** [comment]
closes #1568

**Review by borgmanJeremy — APPROVED** [review]

File: src/utils/screenshotsaver.cpp [hunk 1]
```diff
@@ -25,6 +26,31 @@ ScreenshotSaver::ScreenshotSaver(const unsigned id)
...
+    QImageWriter imageWriter{ &buffer, "PNG" };
+    imageWriter.write(capture.toImage());
+
+    QPixmap pngPixmap;
+    bool isLoaded = pngPixmap.loadFromData(
+      reinterpret_cast<uchar*>(array.data()), array.size(), "PNG");
+    if (isLoaded) {
+        // Need to send message before copying to clipboard
+        SystemNotification().sendMessage(
+          QObject::tr("Capture saved to clipboard"));
```
Comment (alecmev) [inline src/utils/screenshotsaver.cpp:42]: This sends a second notification in both branches of `saveToClipboard`.

**Comment by alecmev** [comment]
@borgmanJeremy Should [this hack](https://github.com/flameshot-org/flameshot/pull/1282) be removed now, perhaps?

**Comment by borgmanJeremy** [comment]
I *think* that fixes a different wayland issue but would need to test it.

**Comment by nullobsi** [comment]
They both refer to the same Qt issue, but testing would definitely have to be done to make sure.

I've made another commit that fixes the double notification, so I open another PR as soon as I can figure out how to get github to work with me

**Pull request merged**